libraries/AP_HAL_ChibiOS/DSP.cpp

```diff
@@ -143,6 +149,20 @@ void DSP::step_hanning(FFTWindowStateARM* fft, const float* samples, uint16_t bu
     TIMER_END(_hanning_timer);
 }
 
+// step 1: filter the incoming samples through a Hanning window
+void DSP::step_hanning(FFTWindowStateARM* fft, FloatBuffer& samples, uint16_t advance)
+{
+    TIMER_START(_hanning_timer);
+    // 5us
+    // apply hanning window to gyro samples and store result in _freq_bins
+    // hanning starts and ends with 0, could be skipped for minor speed improvement
+    samples.peekbytes(&fft->_freq_bins[0], fft->_window_size);
```

Comment:
peekbytes can fail here? Should we care if that happens?

Reply:
ok

Reply:
Not much we can do, I have made it zero out the remainder so that at least we are not processing stale data